app/gui/view/graph-editor/src/builtin/visualization/java_script/table.js

```diff
@@ -95,328 +49,146 @@ class TableVisualization extends Visualization {
-                            content.day,
-                            content.hour,
-                            content.minute,
-                            content.second,
-                            content.nanosecond / 1000000
-                        )
-                        to_render =
-                            js_date.toISOString().substring(0, 19).replace('T', ' ') +
-                            (js_date.getMilliseconds() === 0 ? '' : '.' + js_date.getMilliseconds())
-                        to_render = '<span style="white-space: nowrap;">' + to_render + '</span>'
-                    }
+        function toRender(content) {
+            if (content instanceof Array) {
+                if (isMatrix(content)) {
+                    return `[Vector ${content.length} rows x ${content[0].length} cols]`
+                } else if (isObjectMatrix(content)) {
+                    return `[Vector ${content.length} objects]`
+                } else {
+                    return `[Vector ${content.length} items]`
                 }
-                result += '<td class="plaintext">' + to_render + '</td>'
-            }
-            result += '<tr>'
-            if (parsedData.indices_header) {
-                parsedData.indices_header.forEach(addHeader)
             }
-            parsedData.header.forEach(addHeader)
-            result += '</tr>'
-            let rows = 0
-            if (parsedData.data && parsedData.data.length > 0) {
-                rows = parsedData.data[0].length
-            } else if (parsedData.indices && parsedData.indices.length > 0) {
-                rows = parsedData.indices[0].length
-            }
-            for (let i = 0; i < rows; ++i) {
-                result += '<tr>'
-                if (parsedData.indices) {
-                    parsedData.indices.forEach(ix => addHeader(ix[i]))
+
+            if (content instanceof Object) {
+                const type = content.type
+                if (type === 'BigInt') {
+                    return BigInt(content.value)
+                } else if (type === 'Date') {
+                    return new Date(content.year, content.month - 1, content.day)
+                } else if (type === 'Time_Of_Day') {
+                    return new Date(
+                        0,
+                        0,
+                        1,
+                        content.hour,
+                        content.minute,
+                        content.second,
+                        content.nanosecond / 1000000
+                    )
+                } else if (type === 'Date_Time') {
+                    return new Date(
+                        content.year,
+                        content.month - 1,
+                        content.day,
+                        content.hour,
+                        content.minute,
+                        content.second,
+                        content.nanosecond / 1000000
+                    )
```

Comment:
How do I distinguish in the visualization if I got a `Date` or `Time_Of_Day`? Is Date rendered with `00:00:00.0` and time of day as `01-01-1970 hh:MM:ss`?

I'm not sure what is the current state, but IMO they should be distinguishable. Does keeping it as `Date` give us any advantage in the grid? Because maybe we could just visualize it as formatted string (so time of day would just have the `hh:mm:ss` and Date would not have the hours but just `yyyy-mm-dd`).

Reply:
Also, could you please post a screenshot of how the Dates are visualized?

I have a vague memory of them including a timezone. But here I see that the timezone is _not_ passed when constructing. So if the timezone were displayed it could very well be wrong. We have to either display a correct timezone or not display it at all.